### ISSUE TITLE
uv doesn't include pyparsing in the requirements installed when installing pipx on python 3.10 on linux

### ISSUE THREAD

**Issue by arohrbach-nx** [astral-sh/uv#4098]
On Ubuntu 22.04 under WSL 2, I tried to set up a new python 3.10 environment to run a script with pipx, and uv installed a pipx that crashes because one of its dependencies is missing.

What I did:
1. Follow the instructions at https://github.com/pyenv/pyenv/wiki#suggested-build-environment to make sure I have all the build dependencies for python 3.10
(i.e., I ran 
```sh
sudo apt update; sudo apt install build-essential libssl-dev zlib1g-dev \
libbz2-dev libreadline-dev libsqlite3-dev curl git \
libncursesw5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev
``` 
)

2. I installed python 3.10 and uv into it
```
pyenv install 3.10
pyenv global 3.10
pip install uv
pyenv rehash
```
pip downloaded `uv-0.2.8-py3-none-manylinux_2_17_x86_64.manylinux2014_x86_64.whl`

3. I used uv to install pipx: `uv pip install --system pipx`
4. pipx didn't work until I ran `pip install pipx` because it was missing its dependency pyparsing, which pip installed

Command I invoked, current uv version, and comparsion with pip's output:
```
alan@arohrbac3530-ny:~$ uv pip install --system pipx --verbose
DEBUG Searching for Python interpreter in search path
DEBUG Found CPython 3.10.14 at `/home/alan/.pyenv/versions/3.10.14/bin/python3` (search path)
DEBUG Using Python 3.10.14 environment at .pyenv/versions/3.10.14/bin/python3
DEBUG Acquired lock for `.pyenv/versions/3.10.14`
DEBUG Requirement satisfied: argcomplete>=1.9.4
DEBUG Requirement satisfied: click
DEBUG Requirement satisfied: packaging>=20
DEBUG Requirement satisfied: pipx
DEBUG Requirement satisfied: platformdirs>=2.1
DEBUG Requirement satisfied: tomli ; python_version < '3.11'
DEBUG Requirement satisfied: userpath!=1.9, >=1.6
Audited 1 package in 9ms
alan@arohrbac3530-ny:~$ uv --version
uv 0.2.8
alan@arohrbac3530-ny:~$ pip install pipx
Requirement already satisfied: pipx in ./.pyenv/versions/3.10.14/lib/python3.10/site-packages (1.6.0)
Requirement already satisfied: userpath!=1.9,>=1.6 in ./.pyenv/versions/3.10.14/lib/python3.10/site-packages (from pipx) (1.9.2)
Requirement already satisfied: argcomplete>=1.9.4 in ./.pyenv/versions/3.10.14/lib/python3.10/site-packages (from pipx) (3.3.0)
Requirement already satisfied: platformdirs>=2.1 in ./.local/lib/python3.10/site-packages (from pipx) (2.5.2)
Requirement already satisfied: packaging>=20 in ./.local/lib/python3.10/site-packages (from pipx) (21.3)
Requirement already satisfied: tomli in ./.pyenv/versions/3.10.14/lib/python3.10/site-packages (from pipx) (2.0.1)
Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in ./.pyenv/versions/3.10.14/lib/python3.10/site-packages (from packaging>=20->pipx) (3.1.2)
Requirement already satisfied: click in ./.pyenv/versions/3.10.14/lib/python3.10/site-packages (from userpath!=1.9,>=1.6->pipx) (8.1.7)

```

**Comment by charliermarsh** [comment]
It looks like you have some packages installed in the user directory (presumedly?) and some in the pyenv environment. uv is probably seeing a newer packaging version in the pyenv environment, but pip is seeing the older packaging version in the user environment, and the user environment takes precedence at runtime.

**Comment by charliermarsh** [comment]
I'm referring to the fact that `packaging>=20 in ./.local/lib/python3.10/site-package` comes from `.local` while other dependencies like `pipx in ./.pyenv/versions/3.10.14/lib/python3.10/site-packages` come from `.pyenv`. We don't support the user scheme so not sure what to do here.

**Comment by arohrbach-nx** [comment]
If I do either
```
uv pip uninstall --system $(uv pip freeze)
uv pip install --system pipx
```
outside of a venv, or
```
uv pip uninstall $(uv pip freeze)
uv pip install pipx
```
inside of a venv created with `uv venv -p 3.10 venv-pipx-310`,
I get the same set of packages installed by `uv`:
```
Resolved 7 packages in 15ms
Installed 7 packages in 5ms
 + argcomplete==3.3.0
 + click==8.1.7
 + packaging==24.0
 + pipx==1.6.0
 + platformdirs==4.2.2
 + tomli==2.0.1
 + userpath==1.9.2
```
In the venv, this appears to be fine for `pipx` to run, but out of the venv, `pipx` crashes trying to import pyparsing.
> the user environment takes precedence at runtime.

So I guess `pipx` not in a venv is crashing because at runtime it decides to try to import from the version of `packaging` in .local instead of the newer version that `uv` installed into `.pyenv` (and the one in .local is older and has an extrra, unsatisfied dependency)?

**Comment by arohrbach-nx** [comment]
I guess I don't know what I'd expect `uv` to be able to do here either, unless it was going to add support for the user scheme.
Perhaps what I'm missing is a best practice on where to put tools that I want to have globally available on my development machine.

In some sense, it feels like I really have a pyenv issue, in that a brand new pyenv environment isn't isolated from the user site-packages that will take precedence over anything installed into the pyenv environment's site-packages. 
(Also, it's weird that I ended up with packaging 21.3 in user site-packages but pyparsing 2.4.7 that it depends on must have made into system site-packages for a non-pyenv python 3.10 install, but I guess that's really a pip thing)

One way or another, I found it surprising that I could run `uv pip install --system pipx` and get a broken `pipx` with no warning about it. Maybe `uv` could check user site-packages for packages that will result in incompatibilities and warn about them (which I guess is only a concern with the `--system` flag and not if you're using a venv)?
